### PR TITLE
fix(providers): add reasoning.fieldName support to classic openai provider (Fixes #2505)

### LINKED ISSUE #2488: Support configurable reasoning field name for Ollama (delta.reasoning) via provider-alias ephemeral
## Summary

llxprt only captures streaming reasoning tokens from the OpenAI-style field `delta.reasoning_content`. Ollama's OpenAI-compatible endpoint (`/v1/chat/completions`) emits reasoning under a different key: `delta.reasoning`. As a result, when a reasoning model is served through Ollama, the thinking trace is silently dropped in the streaming path and never displayed, even though the model is reasoning correctly.

We should make the reasoning source field name configurable per provider (via a provider-alias ephemeral setting), defaulting to the current standard `reasoning_content`, so users can point llxprt at Ollama (or any nonstandard backend) and surface reasoning without code changes.

## Evidence

Observed against a local Ollama serving a reasoning model. The streaming delta keys emitted were:

    delta keys: { "role", "content", "reasoning" }

llxprt's capture reads only `reasoning_content`:

- packages/providers/src/openai-vercel/vercelReasoningCapture.ts:48 - type declares delta as reasoning_content only
- packages/providers/src/openai-vercel/vercelReasoningCapture.ts:60 - reads parsed.choices[0].delta.reasoning_content
- packages/providers/src/openai-vercel/vercelStreamProcessor.ts - emits with sourceField reasoning_content

Note: the non-streaming path already reads result.reasoning (packages/providers/src/openai-vercel/vercelNonStreamingResponse.ts:79), so non-streaming can surface it while streaming cannot. This is an inconsistency worth aligning.

Field name by backend:

| Backend | Reasoning field in delta |
|---|---|
| OpenAI / LM Studio / vLLM / DeepSeek | reasoning_content |
| Ollama | reasoning |

## Proposed solution

1. Add a new provider-alias ephemeral setting, e.g. reasoning.field_name (default: reasoning_content). This sits alongside the existing reasoning ephemeral group (enabled / includeInResponse / includeInContext / stripFromContext).
2. In the streaming capture (vercelReasoningCapture.ts) and stream processor, read the configured field name; fall back to reasoning_content, then reasoning, if unset.
3. Ensure the setting flows from the provider alias config into the capture path.
4. Keep non-streaming and streaming behavior consistent.

## Acceptance criteria

- A provider alias can set the reasoning field name (default unchanged: reasoning_content).
- With the field set to reasoning, a reasoning model served via Ollama surfaces its thinking in llxprt streaming output.
- Existing providers (LM Studio, vLLM, OpenAI) are unaffected when the setting is left at default.
- Streaming and non-streaming paths behave consistently.

## Notes / scope

- This is a provider-support concern (transport field-name shape), not model-specific. Ornith via vLLM already works because vLLM emits reasoning_content; this is specifically to support Ollama and any other backend using a nonstandard key.

### LINKED ISSUE #2505: Classic openai provider drops Ollama reasoning: needs reasoning.fieldName support like openai-vercel (follow-up to #2488)
## Summary

PR #2491 (Fixes #2488) added a configurable reasoning field name (reasoning.fieldName, with auto-fallback to delta.reasoning for Ollama) so that reasoning models served via Ollama surface their thinking in streaming output. However, that fix was only applied to the **openai-vercel** provider. The **classic openai provider** was not patched and still hardcodes delta.reasoning_content, so Ollama reasoning is still silently dropped when using --provider openai.

This is the same underlying issue as #2488, but for the classic OpenAI provider code path.

## Evidence

Reproduced against a local Ollama serving a reasoning model (North-Mini-Code) at http://127.0.0.1:11434/v1/.

Raw Ollama streaming deltas emit reasoning under delta.reasoning (confirmed on the wire), with the polished answer under delta.content.

- With --provider openai-vercel and --set reasoning.fieldName=reasoning: thinking IS displayed. (PR #2491 path works.)
- With --provider openai (classic) and the same settings: thinking is NOT displayed. Debug log shows namespace llxprt:provider:openai with cleanedReasoningLength=0, reasoningLength=0, hasThinking=false.

The classic provider hardcodes the field name:

- packages/providers/src/openai/OpenAIResponseParser.js:310 - const reasoningContent = delta.reasoning_content;
- packages/providers/src/openai/OpenAIResponseParser.js:331 - sourceField: 'reasoning_content'
- packages/providers/src/openai/OpenAIStreamProcessor.js:366 - sourceField: 'reasoning_content'
- packages/providers/src/openai/parseResponsesStream.js:45,134 - sourceField: 'reasoning_content'

There is no read of the reasoning.fieldName ephemeral anywhere under packages/providers/src/openai/ (grep for reasoning_content, delta.reasoning, and fieldName confirms only reasoning_content is referenced).

## Proposed solution

Apply the same reasoning.fieldName handling from PR #2491 to the classic openai provider:

1. Read the reasoning.fieldName ephemeral (default reasoning_content) in the classic streaming parse path (OpenAIResponseParser / OpenAIStreamProcessor).
2. When the setting is unset, auto-fall-back to also check delta.reasoning (Ollama compatibility), matching the openai-vercel behavior.
3. When explicitly set, only read the configured field.
4. Propagate the actual field name used to the ThinkingBlock.sourceField metadata (provenance), as PR #2491 did for the vercel path.
5. Keep existing providers (LM Studio, vLLM, OpenAI) unchanged by default.

## Acceptance criteria

- With --provider openai (classic), --set reasoning.fieldName=reasoning, pointed at an Ollama reasoning model, the thinking trace is displayed in streaming output.
- With the setting unset, both reasoning_content and reasoning are honored (auto-fallback), so both standard providers and Ollama work out of the box.
- Standard providers (reasoning_content) are unaffected.
- sourceField provenance reflects the actual captured field.

## Guardrails

Do not loosen lint/complexity rules and do not add eslint-disable / ts-ignore / ts-expect-error suppressions. Fix the underlying issue rather than silencing rules.

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Fixes #2505.

PR #2491 (Fixes #2488) added a configurable reasoning field name (`reasoning.fieldName`, with auto-fallback to `delta.reasoning` for Ollama) to the **openai-vercel** provider so reasoning models served via Ollama surface their thinking in streaming output. However, that fix was **not** applied to the **classic openai provider** (`packages/providers/src/openai/`), which still hardcoded `delta.reasoning_content`. As a result, Ollama reasoning (emitted under `delta.reasoning`) was silently dropped when using `--provider openai`.

This PR applies the same `reasoning.fieldName` handling to the classic openai provider, mirroring the proven pattern from PR #2491.

## Changes

- **`OpenAIResponseParser.ts`**: `parseStreamingReasoningDelta` now accepts an optional `fieldName`. Added `readReasoningFromDelta` and `resolveReasoningValue` helpers that read the configured field (default `reasoning_content`) and, when the setting is unset, auto-fall-back to `delta.reasoning` (Ollama). The actual captured field name is propagated as the `ThinkingBlock.sourceField` for provenance. Empty/whitespace-only field names are normalized to "unset" so auto-fallback still applies.
- **`OpenAIStreamProcessor.ts`**: Added `reasoningFieldName?` to `StreamProcessorDeps`, threaded it to `parseStreamingReasoningDelta`, and track the actual source field per-chunk in `StreamingState.reasoningSourceField`. The terminal combined thinking block now uses the tracked source field instead of a hardcoded value.
- **`OpenAIStreamProcessorState.ts`**: Added `reasoningSourceField` to `StreamingState`.
- **`OpenAIProvider.ts`**: Reads `options.settings.get('reasoning.fieldName')` and threads it through `dispatchResponse` into the streaming deps.

## Behavior

| Setting | `delta.reasoning_content` | `delta.reasoning` | Result |
|---|---|---|---|
| unset | present | — | captured, sourceField=`reasoning_content` |
| unset | absent | present | captured (auto-fallback), sourceField=`reasoning` |
| unset | present | present | `reasoning_content` wins |
| `reasoning` | — | present | captured, sourceField=`reasoning` |
| `reasoning` | present | — | ignored (explicit field not present) |
| `reasoning_content` | absent | present | ignored (no auto-fallback when explicit) |

Standard providers (LM Studio, vLLM, OpenAI) that emit `reasoning_content` are unaffected by default.

## Tests

- `OpenAIResponseParser.fieldName.test.ts` (new, 13 tests): unit-level behavioral tests for the configurable field name, auto-fallback, precedence, edge cases (null/non-string/empty primary values), and empty/whitespace field-name normalization.
- `OpenAIProviders.fieldName.test.ts` (new, 6 tests): integration tests exercising `processStreamingResponse` end-to-end, verifying the wiring (`reasoningFieldName` → parser → terminal `ThinkingBlock.sourceField`) and multi-chunk accumulation.
- All existing openai provider tests (777) continue to pass.

## Verification

- `npm run typecheck` — pass
- `npm run lint` (changed files) + `check-eslint-guard.ts` — pass (no rules loosened, no suppressions)
- `npm run format` — clean
- `npm run test` — all packages pass (5135+ tests, 0 failures)
- `npm run build` — clean

## Review gates

Open Code Review (ocr) run with `--timeout 20`; both findings addressed (empty-string field-name normalization + tests). Code review by typescriptreviewer subagent; all Medium/Low findings addressed (mixed-source provenance uses last-field-wins matching Vercel, type assertion removed, fallback now triggers on non-usable primary values, integration tests added).

### CODE REVIEW

File: packages/providers/src/openai/OpenAIProviders.fieldName.test.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    model: 'test-model',
+    choices: [
+      {
+        index: 0,
+        delta,
+        finish_reason: finishReason,
+      },
+    ],
+  };
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIProviders.fieldName.test.ts:58]: Addressed: switched to a deterministic fixed timestamp (`created: 1000000`) in commit 5cd89e2cb so the chunk data is stable for any future snapshot or field-level assertions.

File: packages/providers/src/openai/OpenAIProviders.fieldName.test.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    toolCallPipeline: new ToolCallPipeline(),
+    textToolParser: {
+      parse: (text: string) => ({ toolCalls: [], cleanedContent: text }),
+    },
+    logger: {
+      debug: vi.fn(),
+      warn: vi.fn(),
+      log: vi.fn(),
+      error: vi.fn(),
+    },
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIProviders.fieldName.test.ts:73]: Acknowledged — the OCR analysis itself confirms the mocks are complete and there is no issue here, so no change needed.

File: packages/providers/src/openai/OpenAIProviders.fieldName.test.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      {} as Parameters<typeof processStreamingResponse>[4],
+      [],
+      {} as Parameters<typeof processStreamingResponse>[6],
+      undefined,
+      undefined,
+      makeDeps(reasoningFieldName),
+      async function* () {
+        yield* [] as IContent[];
+      } as Parameters<typeof processStreamingResponse>[10],
+    ),
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIProviders.fieldName.test.ts:100]: Valid maintainability note. The positional `Parameters<typeof processStreamingResponse>[N]` indexing is deliberate here because it mirrors the established convention used in the sibling test file OpenAIProviders.issue1844.test.ts (the repo's existing pattern for exercising processStreamingResponse directly). Keeping the two files consistent aids grep-ability and review; diverging to a named-wrapper helper in this one test would introduce an inconsistency. The function signature is stable and any signature change would surface across both test files together. Leaving as-is to preserve convention consistency.

File: packages/providers/src/openai/OpenAIResponseParser.ts [hunk 4]
```diff
@@ -33,8 +33,63 @@ const CODE_FENCE_END = new RegExp(CODE_FENCE_END_SOURCE, 'm');
...
+function readReasoningFromDelta(
+  delta: DeltaWithReasoning,
+  fieldName: string | undefined,
+): { content: string; actualFieldName: string } | undefined {
+  // Treat empty/whitespace-only field names as unset so auto-fallback still
+  // applies (issue #2505: guards against reasoning.fieldName='' misconfig).
+  const normalizedFieldName =
+    fieldName !== undefined && fieldName.trim().length > 0
+      ? fieldName
+      : undefined;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIResponseParser.ts]: Addressed in commit f7a10434a: the field name is now trimmed before being used as the delta property key, so a whitespace-padded value like '  reasoning  ' resolves to delta.reasoning correctly. Added a test covering this case.

File: packages/providers/src/openai/OpenAIResponseParser.ts [hunk 5]
```diff
@@ -33,8 +33,63 @@ const CODE_FENCE_END = new RegExp(CODE_FENCE_END_SOURCE, 'm');
...
+  fieldName: string | undefined,
+): { value: unknown; actualFieldName: string } {
+  if (hasUsableReasoning(primaryValue)) {
+    return { value: primaryValue, actualFieldName: explicitField };
+  }
+  if (fieldName === undefined && hasUsableReasoning(ollamaReasoning)) {
+    return { value: ollamaReasoning, actualFieldName: 'reasoning' };
+  }
+  return { value: primaryValue, actualFieldName: explicitField };
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIResponseParser.ts:58]: Addressed in commit f7a10434a: resolveReasoningValue now returns undefined when no usable value or fallback applies, making the contract explicit. The returned type is narrowed to { value: string; actualFieldName: string } | undefined so callers no longer need to filter non-usable values. Added a test for the non-usable explicit-field case.

File: packages/providers/src/openai/OpenAIProviders.fieldName.test.ts [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+  );
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIProviders.fieldName.test.ts:102]: Acknowledged. This positional-indexing pattern (Parameters<typeof processStreamingResponse>[N]) is used deliberately to match the established convention in the sibling OpenAIProviders.issue1844.test.ts, which exercises the same function with the identical pattern. Keeping the two files consistent aids cross-file review and grep-ability; introducing a named-wrapper helper here would diverge from the existing repo convention. The function signature is stable and any signature change would surface across both test files simultaneously.

File: packages/providers/src/openai/OpenAIProviders.fieldName.test.ts [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    logger: {
+      debug: vi.fn(),
+      warn: vi.fn(),
+      log: vi.fn(),
+      error: vi.fn(),
+    },
+    getBaseURL: () => undefined,
+    reasoningFieldName,
+  } as unknown as StreamDeps;
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIProviders.fieldName.test.ts:77]: Acknowledged. The 'as unknown as StreamDeps' double cast mirrors the established convention in OpenAIProviders.issue1844.test.ts (same 'as unknown as Parameters<typeof processStreamingResponse>[9]' pattern). Typing the mock object literal directly as StreamDeps would require fully implementing every method of the GemmaToolCallParser class and DebugLogger, which is heavier than the test needs and diverges from the existing repo test style. The partial mock is sufficient because the streaming path under test only calls logger.debug/warn/log/error and textToolParser.parse, all of which are provided. Keeping consistency with the existing test file.

File: packages/providers/src/openai/OpenAIProviders.fieldName.test.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIProviders.fieldName.test.ts:102]: Acknowledged. The client is passed as {} (an empty object cast) which matches the established convention in the sibling OpenAIProviders.issue1844.test.ts (same 'as Parameters<typeof processStreamingResponse>[6]' pattern). This is safe for these tests because they emit reasoning-only chunks with no tool calls: handleToolCallsWithoutText short-circuits via hasToolsButNoTextContent returning false before client is ever dereferenced. The continuation path is exercised by the dedicated issue1844 tests.

[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai/OpenAIProviders.fieldName.test.ts:102]: Acknowledged — same as the earlier note. This positional-indexing pattern intentionally mirrors the established convention in OpenAIProviders.issue1844.test.ts for consistency and grep-ability across the two sibling test files. The function signature is stable and any change would surface across both files simultaneously.
